src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/QualityScoreDistributionIntegrationTest.java

```diff
@@ -37,4 +40,109 @@ public void test() throws IOException {
         Assert.assertTrue(pdf.exists());
         IntegrationTestSpec.assertEqualTextFiles(outfile, expectedFile, "#");
     }
+
+    @Test(groups = {"R"})
+    public void testWithOQs() throws IOException {
+        final File input = new File(TEST_DATA_DIR, "originalQuals.chr1.1-1K.bam");
+        final File expectedFile = new File(TEST_DATA_DIR, "originalQuals.chr1.1-1K.QualityScoreDistribution.txt");
+        final File outfile = BaseTest.createTempFile("testQualityScoreDistribution.originalQuals", ".metrics");
+        final File pdf = BaseTest.createTempFile("testQualityScoreDistribution.originalQuals", ".pdf");
+        final String[] args = {
+                "--INPUT", input.getAbsolutePath(),
+                "--OUTPUT", outfile.getAbsolutePath(),
+                "--CHART", pdf.getAbsolutePath(),
+                "--PRODUCE_PLOT", "true",
+        };
+        runCommandLine(args);
+
+        try (final FileReader actualReader = new FileReader(outfile);) {
+            final MetricsFile<?, Integer> output = new MetricsFile<>();
+            output.read(actualReader);
+            Assert.assertEquals(output.getAllHistograms().size(), 2);
+            Assert.assertEquals(output.getHistogram().size(), 36);
+        }
+        Assert.assertTrue(pdf.exists());
+        IntegrationTestSpec.assertEqualTextFiles(outfile, expectedFile, "#");
+    }
+
+
+    @Test(groups = {"R"})
+    public void test_PF_READS_ONLY_false() throws IOException {
+        final File input = new File(TEST_DATA_DIR, "example_pfFail_reads.bam");
+        final File expectedFile = new File(TEST_DATA_DIR, "pfFailBam.pf.txt");
+        final File outfile = BaseTest.createTempFile("pfFailBam.pf", ".metrics");
+        final File pdf = BaseTest.createTempFile("pfFailBam.pf", ".pdf");
+        outfile.deleteOnExit();
+        pdf.deleteOnExit();
+        final String[] args = {
+                "--INPUT", input.getAbsolutePath(),
+                "--OUTPUT", outfile.getAbsolutePath(),
+                "--CHART", pdf.getAbsolutePath(),
+                "--PF_READS_ONLY", "false",
+                "--PRODUCE_PLOT", "true",
+        };
```

Comment:
I see the main checking is happening with IntegrationTestSpec.assertEqualTextFiles. This set of tests looks like a good candidate for a data provider.